src/Microsoft.PowerApps.TestEngine/System/FileSystem.cs

```diff
@@ -1,91 +1,230 @@
+// Licensed under the MIT license.
+
+using System.ComponentModel.Composition;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.PowerApps.TestEngine.System
+{
+    /// <summary>
+    /// Wrapper for any System.IO methods needed
+    /// </summary>
+    [Export(typeof(IFileSystem))]
+    public class FileSystem : IFileSystem
+    {
+        public void CreateDirectory(string directoryName)
+        {
+            if (IsNonUNCDirectoryPath(directoryName))
+            {
+                Directory.CreateDirectory(directoryName);
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Path invalid or write to path: {0} not permitted.", directoryName));
+            }
+        }
+
+        public bool Exists(string directoryName)
+        {
+            if (IsNonUNCDirectoryPath(directoryName))
+            {
+                return Directory.Exists(directoryName);
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Path invalid or write to path: {0} not permitted.", directoryName));
+            }
+        }
+
+        public bool FileExists(string fileName)
+        {
+            if (IsValidFilePath(fileName))
+            {
+                return File.Exists(fileName);
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Invalid file path {0}.", fileName));
+            }
+        }
+
+        public string[] GetFiles(string directoryName)
+        {
+            if (IsNonUNCDirectoryPath(directoryName))
+            {
+                return Directory.GetFiles(directoryName);
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Path invalid or read from path: {0} not permitted.", directoryName));
+            }
+        }
+
+        public string[] GetFiles(string directoryName, string searchPattern)
+        {
+            if (IsNonUNCDirectoryPath(directoryName))
+            {
+                return Directory.GetFiles(directoryName, searchPattern);
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Path invalid or read from path: {0} not permitted.", directoryName));
+            }
+        }
+
+        public void WriteTextToFile(string filePath, string text)
+        {
+            if (IsWritePermittedFilePath(filePath))
+            {
+                if (File.Exists(filePath))
+                {
+                    File.AppendAllText(filePath, text);
+                }
+                else
+                {
+                    File.WriteAllText(filePath, text);
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Write to path: {0} not permitted, ensure path is rooted in base TestEngine path.", filePath));

```

Comment:
Instead of 
      "Write to path: { 0 } ....
Do
      "Write to path \'{0}\' ...
For consistency with other errors.

Reply:
added